misc/bazel/workspace.bzl

```diff
@@ -0,0 +1,38 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
+
+def _ql_utils_impl(repository_ctx):
+    root = repository_ctx.path(Label("//:WORKSPACE.bazel")).realpath.dirname
+    repository_ctx.file("BUILD.bazel")
+    repository_ctx.template(
+        "source_dir.bzl",
+        Label("@ql//misc/bazel:source_dir.bzl.tpl"),
+        substitutions = {"{root}": str(root)},
+    )
+
+_ql_utils = repository_rule(
+    implementation = _ql_utils_impl,
+)
+
+def ql_workspace():
+    _ql_utils(name = "ql_utils")
+
+    maybe(
+        repo_rule = http_archive,
+        name = "rules_pkg",
+        sha256 = "62eeb544ff1ef41d786e329e1536c1d541bb9bcad27ae984d57f18f314018e66",
+        urls = [
+            "https://mirror.bazel.build/github.com/bazelbuild/rules_pkg/releases/download/0.6.0/rules_pkg-0.6.0.tar.gz",
+            "https://github.com/bazelbuild/rules_pkg/releases/download/0.6.0/rules_pkg-0.6.0.tar.gz",
+        ],
+    )
+
+    maybe(
+        repo_rule = http_archive,
+        name = "platforms",
```

Comment:
Do we need to pull this in?
If we can avoid it, we'd not need to coordinate the bump cross-submodules when we bump it in semmle-code

Reply:
in semmle-code this is executed after semmle-code itself pulls those in, so the versions specified in semmle-code have priority over these, and semmle-code can change them without needing a submodule bump. Did I just write semmle-code way too many times? :smile:

Reply:
Ah, that makes sense. I guess for consistency we should still bump those together, but then it's not too important to do so at the precise same time :)

Reply:
also, at some point we will be able to use [bzlmod](https://docs.bazel.build/versions/5.0.0/bzlmod.html) and this will be so, so much better :slightly_smiling_face: